core/src/test/scala/org/apache/spark/memory/MemoryManagerSuite.scala

```diff
@@ -335,6 +335,28 @@ private[memory] trait MemoryManagerSuite extends SparkFunSuite {
     tMemManager.releaseExecutionMemory(500L, c)
     assert(tMemManager.getMemoryConsumptionForThisTask === 0L)
   }
+
+  test("task peak execution memory usage") {
+    val memoryManager = createMemoryManager(
+      maxOnHeapExecutionMemory = 1000L,
+      maxOffHeapExecutionMemory = 1000L)
+
+    val tMemManager = new TaskMemoryManager(memoryManager, 1)
+    val offHeapConsumer = new TestMemoryConsumer(tMemManager, MemoryMode.OFF_HEAP)
+    val onHeapConsumer = new TestMemoryConsumer(tMemManager, MemoryMode.ON_HEAP)
+
+    val result1 = Future {
```

Comment:
Do we actually need to use Futures in this test case? I see that many of the existing nearby test cases are also doing this, but I'm wondering whether perhaps those existing tests might have just copied it from some other test. If we don't actually require asynchrony in this test case, maybe we could eliminate it for the sake of simplicity (and perhaps test flakiness avoidance)?

Reply:
I think we don't really need this in term of the semantics of this test case. Made it synchronous.